src/ceph.in

```diff
@@ -169,6 +169,8 @@ else:
 
 def raw_write(buf):
     sys.stdout.flush()
+    if isinstance(buf, str):
+        buf = buf.encode()
```

Comment:
see http://pulpito.ceph.com/kchai-2018-11-25_10:39:02-rados-wip-kefu2-testing-2018-11-22-1635-distro-basic-smithi/3282173/

```
2018-11-25T11:18:42.634 INFO:tasks.workunit.client.0.smithi017.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/crush/crush-choose-args.sh:56: TEST_choose_args_ceph osd get
crushmap
2018-11-25T11:18:43.053 INFO:tasks.workunit.client.0.smithi017.stderr:2
2018-11-25T11:18:43.299 INFO:tasks.workunit.client.0.smithi017.stderr:Traceback (most recent call last):
2018-11-25T11:18:43.299 INFO:tasks.workunit.client.0.smithi017.stderr:  File "/usr/bin/ceph", line 1252, in <module>
2018-11-25T11:18:43.300 INFO:tasks.workunit.client.0.smithi017.stderr:    retval = main()
2018-11-25T11:18:43.300 INFO:tasks.workunit.client.0.smithi017.stderr:  File "/usr/bin/ceph", line 1231, in main
2018-11-25T11:18:43.300 INFO:tasks.workunit.client.0.smithi017.stderr:    raw_write(outbuf)
2018-11-25T11:18:43.300 INFO:tasks.workunit.client.0.smithi017.stderr:  File "/usr/bin/ceph", line 173, in raw_write
2018-11-25T11:18:43.300 INFO:tasks.workunit.client.0.smithi017.stderr:    buf = buf.encode()
2018-11-25T11:18:43.300 INFO:tasks.workunit.client.0.smithi017.stderr:UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 20: ordinal not in range(128)
2018-11-25T11:18:43.309 INFO:tasks.workunit.client.0.smithi017.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/crush/crush-choose-args.sh:56: TEST_choose_args_return 1
2018-11-25T11:18:43.310 INFO:tasks.workunit.client.0.smithi017.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/crush/crush-choose-args.sh:40: run:  return 1
```

so we cannot use ASCII codec for a random binary string. probably we should not write `bytes` to `sys.__stdout__` or `sys.stdout`. because 

- in python2, `sys.stdout` is a "file". so we can just write `str` to it.
- in python3, `sys.stdout` is a "io.TextIOWrapper(name='<stdout>', mode='w', encoding='UTF-8'>)" around a binary stream. so it's accept text.

so, i'm inclined to keep it as it is.

Reply:
@tchaikov Python throws a runtime error at exit --- hence the desire to fix it.

Reply:
sorry, should have read the commit message. will try to reproduce it and take a closer look at it.

Reply:
@dillaman what was the runtime error thrown at exit? i cannot reproduce it with `cmake -DWITH_PYTHON3=ON`. and i can
```sh
$ head -n1 bin/ceph
#!/usr/bin/python3
$bin/ceph osd getcrushmap > /tmp/foo
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
2018-11-26 14:34:56.877 7f5e64d56700 -1 WARNING: all dangerous and experimental features are enabled.
2018-11-26 14:34:56.905 7f5e64d56700 -1 WARNING: all dangerous and experimental features are enabled.
7
```
just fine.

Reply:
@tchaikov It might have been in CLI mode

Reply:
@dillaman could you define "CLI mode"? 

```
$ bin/ceph -s
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
2018-11-26 21:34:14.253 7f5de4207700 -1 WARNING: all dangerous and experimental features are enabled.
2018-11-26 21:34:14.469 7f5de2754700 -1 WARNING: all dangerous and experimental features are enabled.
  cluster:
    id:     6bd2b252-6668-4dc7-9b62-7f9b37841abd
    health: HEALTH_OK

  services:
    mon: 1 daemons, quorum a (age 25s)
    mgr: x(active, since 16s)
    osd: 3 osds: 3 up (since 5s), 3 in (since 5s)

  data:
    pools:   0 pools, 0 pgs
    objects: 0 objects, 0 B
    usage:   60 KiB used, 3.0 GiB / 3 GiB avail
    pgs:
````

also works fine.

Reply:
Perhaps I should have said interactive — type “bin/ceph” and enter